c7n/actions.py

```diff
@@ -276,6 +277,10 @@ def get_groups(self, resources, metadata_key=None):
             # Parse remove_groups
             if remove_target_group_ids == 'matched':
                 remove_groups = r.get('c7n:matched-security-groups', ())
+            elif remove_target_group_ids == 'network-location':
+                for reason in r.get('c7n:NetworkLocation', ()):
+                    if reason['reason'] == 'SecurityGroupMismatch':
+                        remove_groups = [sg for sg in reason['security-groups']]
```

Comment:
the filter implementation is using a dict for the annotation now,  so this usage is invalid now, given that it seems like this functionality should have a unit test.